Firestore/Source/API/FIRDocumentSnapshot.mm

```diff
@@ -40,6 +39,21 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/** Converts a public FIRServerTimestampBehavior into its internal equivalent. */
+static FSTServerTimestampBehavior InternalServerTimestampBehavor(
+    FIRServerTimestampBehavior behavior) {
+  switch (behavior) {
+    case FIRServerTimestampBehaviorNone:
+      return FSTServerTimestampBehaviorNone;
+    case FIRServerTimestampBehaviorEstimate:
+      return FSTServerTimestampBehaviorEstimate;
+    case FIRServerTimestampBehaviorPrevious:
+      return FSTServerTimestampBehaviorPrevious;
+    default:
+      FIREBASE_ASSERT_MESSAGE(false, "Unexpected server timestamp option: %ld", (long)behavior);
```

Comment:
I guess:
1) You're using FIREBASE_ASSERT_MESSAGE instead of FSTFail() just to push us in the direction of the C++ port?
2) There's no FSTFail(...) equivalent so we have to assert(false, ...) instead?  [I'd advocate for adding a FIREBASE_FAIL_MESSAGE() or something at some point, but probably not in this PR].

Feel free to take no action.

Reply:
Yeah, I think the `FIREBASE_ASSERT` macros are kinda busted. There's a bunch of flexibility in there we don't need and there are cases we want that aren't handled cleanly.

Let me take this on as a follow-up.

Reply:
SGTM, or feel free to just open a bug for tracking.